core/src/main/java/com/sequenceiq/cloudbreak/service/image/catalog/VersionBasedImageProvider.java

```diff
@@ -73,22 +77,25 @@ public StatedImages getImages(CloudbreakImageCatalogV3 imageCatalogV3, ImageFilt
 
         List<Image> baseImages = filterImagesByPlatforms(imageFilter.getPlatforms(), imageCatalogV3.getImages().getBaseImages(), vMImageUUIDs);
         List<Image> cdhImages = filterImagesByPlatforms(imageFilter.getPlatforms(), imageCatalogV3.getImages().getCdhImages(), vMImageUUIDs);
+        List<Image> freeipaImages = filterImagesByPlatforms(imageFilter.getPlatforms(), imageCatalogV3.getImages().getFreeIpaImages(), vMImageUUIDs);
 
         List<Image> defaultImages = defaultVMImageUUIDs.stream()
                 .map(imageId -> getImage(imageId, imageCatalogV3.getImages()))
                 .flatMap(Optional::stream)
                 .collect(Collectors.toList());
 
         Collection<String> latestDefaultImageUuids = latestDefaultImageUuidProvider.getLatestDefaultImageUuids(imageFilter.getPlatforms(), defaultImages);
-        Stream.of(baseImages.stream(), cdhImages.stream())
+        (!freeipaImages.isEmpty() ?
+                Stream.of(freeipaImages.stream()) :
```

Comment:
Minor: you don't need to wrap` freeipaImage.stream()` with `Stream.of()`

Reply:
Stream.of wrapping is removed.